numpy/core/src/multiarray/ctors.c

```diff
@@ -2968,11 +2968,25 @@ PyArray_Arange(double start, double stop, double step, int type_num)
     PyArray_ArrFuncs *funcs;
     PyObject *obj;
     int ret;
+    double delta, tmp_len;
     NPY_BEGIN_THREADS_DEF;
 
-    length = _arange_safe_ceil_to_intp((stop - start)/step);
-    if (error_converting(length)) {
-        return NULL;
+    delta = stop - start;
+    tmp_len = delta/step;
+
+    if (tmp_len == 0.0 && delta != 0.0) {
```

Comment:
This needs a comment explaining why it is a special case

Isn't this only satisfied if `step` is `inf` anyway?

Reply:
@eric-wieser This considers som underflow cases.
```python
>>> m = np.float32(3.40282347E+38)
>>> e = np.float32(1.401298E-45)
>>> m
3.4028235e+38
>>> e
1.4012985e-45
>>> e/m
0.0
>>> np.arange(np.float32(0), e, m)
array([], dtype=float64)
```